lib/galaxy/files/sources/invenio.py

```diff
@@ -207,17 +217,37 @@ def user_records_url(self) -> str:
     def to_plugin_uri(self, record_id: str, filename: Optional[str] = None) -> str:
         return f"{self.plugin.get_uri_root()}/{record_id}{f'/{filename}' if filename else ''}"
 
-    def get_records(self, writeable: bool, user_context: OptionalUserContext = None) -> List[RemoteDirectory]:
+    def get_records(
+        self,
+        writeable: bool,
+        user_context: OptionalUserContext = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        query: Optional[str] = None,
+        sort_by: Optional[str] = None,
+    ) -> Tuple[List[RemoteDirectory], int]:
         params: Dict[str, Any] = {}
         request_url = self.records_url
         if writeable:
             # Only draft records owned by the user can be written to.
             params["is_published"] = "false"
             request_url = self.user_records_url
-        # TODO: Add pagination support to FileSources. This is limited to 100 records by default for now.
-        params["size"] = 100
+        size, page = self._to_size_page(limit, offset)
+        params["size"] = size
+        params["page"] = page
+        if query:
+            params["q"] = query
+            params["sort"] = "bestmatch"
```

Comment:
This is to improve search results when [querying the Invenio RDM API](https://inveniordm.docs.cern.ch/reference/rest_api_drafts_records/#search-records) otherwise it will use `"newest"` by default.